ams/server/src/main/java/com/netease/arctic/server/dashboard/PaimonTableDescriptor.java

```diff
@@ -40,7 +56,73 @@ public List<TableFormat> supportFormat() {
 
   @Override
   public ServerTableMeta getTableDetail(AmoroTable<?> amoroTable) {
-    throw new UnsupportedOperationException();
+    FileStoreTable table = getTable(amoroTable);
+    AbstractFileStore<?> store = (AbstractFileStore<?>) table.store();
+
+    ServerTableMeta serverTableMeta = new ServerTableMeta();
+    serverTableMeta.setTableIdentifier(amoroTable.id());
+    serverTableMeta.setTableType(amoroTable.format().name());
+
+    //schema
+    serverTableMeta.setSchema(
+        table.rowType().getFields().stream()
+            .map(s -> new AMSColumnInfo(s.name(), s.type().asSQLString(), !s.type().isNullable(), s.description()))
+            .collect(Collectors.toList())
+    );
+
+    //pk
+    Set<String> primaryKeyNames = new HashSet<>(table.primaryKeys());
+    List<AMSColumnInfo> primaryKeys = serverTableMeta.getSchema()
+        .stream()
+        .filter(s -> primaryKeyNames.contains(s.getField()))
+        .collect(Collectors.toList());
+    serverTableMeta.setPkList(primaryKeys);
+
+    //partition
+    List<AMSPartitionField> partitionFields = store.partitionType()
+        .getFields().stream()
+        .map(f -> new AMSPartitionField(f.name(), null, null, f.id(), null))
+        .collect(Collectors.toList());
+    serverTableMeta.setPartitionColumnList(partitionFields);
+
+    //properties
+    serverTableMeta.setProperties(table.options());
+
+    Map<String, Object> tableSummary = new HashMap<>();
+    Map<String, Object> baseSummary = new HashMap<>();
+    //table summery
+    tableSummary.put("tableFormat", AmsUtil.formatString(amoroTable.format().name()));
+    Snapshot snapshot = store.snapshotManager().latestSnapshot();
+    if (snapshot != null) {
+      TransactionsOfTable transactionsOfTable =
+          manifestListInfo(store, snapshot, (m, s) -> s.dataManifests(m));
+      tableSummary.put("size", AmsUtil.byteToXB(transactionsOfTable.fileSize));
```

Comment:
Although the fields of `TransactionsOfTable` is public, I don't think it's a good practice to using these fields directly instead of using the `getter`.

Maybe we should change these fields to private.

Reply:
yeah